app/Http/Controllers/DeletedWikiMetricsController.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Wiki;
+use App\WikiManager;
+use Carbon\CarbonImmutable;
+use Illuminate\Http\Request;
+
+class DeletedWikiMetricsController extends Controller
+{
+
+    private string $fileName="deleted_wiki_metric.csv";
+
+    /**
+     * Produce a downloadable csv file with deleted wiki metrics.
+     */
+    public function index(Request $request)
+    {
+        $allDeletedWikis = Wiki::onlyTrashed()->get();
+        $output = $this->createOutput($allDeletedWikis);
+        return $this->returnCsv($output);
+    }
+
+    private function returnCsv( $output ) {
+        ob_start();
+        $handle = fopen('php://output', 'r+');
+        fputcsv($handle, [
+            'domain_name_for_wiki',
+            'wiki_deletion_reason',
+            'number_of_wikibases_owned_by_owners_of_this_wiki',
+            'number_of_entities_for_wiki',
+            'number_of_wiki_edits_for_wiki',
+            'number_of_wiki_pages_for_wiki',
+            'number_of_users_for_wiki',
+            'wiki_creation_time',
+            'wiki_deletion_time',
+        ]);
+        foreach ($output as $deletedWikiMetrics) {
+            fputcsv($handle, array_values($deletedWikiMetrics));
+        }
+        $csv = ob_get_clean();
+        return response($csv, 200, [
+            'Content-Type' => 'text/csv',
+            'Content-Disposition' => 'attachment;filename='.CarbonImmutable::now()->toIso8601String().'-'.$this->fileName ]);
+    }
+
+    // TODO: This function should be private. It is right now public and only used for unit tests
```

Comment:
I put up a PR that might help with this: https://github.com/wbstack/api/pull/840

Why don't you take a look and give it a review of your own. If you like it feel free to just merge into this branch that it's currently targetting